src/Solvers/discrete_transforms.jl

```diff
@@ -80,7 +81,7 @@ end
 NoTransform() = DiscreteTransform([nothing for _ in fieldnames(DiscreteTransform)]...)
 
 function DiscreteTransform(plan, direction, grid, dims)
-    arch = architecture(grid)
+    arch = child_architecture(architecture(grid)) # In case we are doing it on a DistributedGrid
```

Comment:
```suggestion
    arch = child_architecture(grid) # In case we are doing it on a DistributedGrid
```

you may want to implement this